src/dotnet.ts

```diff
@@ -1,34 +1,44 @@
 import { execFile } from 'child_process';
 import { promisify } from 'util';
+import * as fs from 'fs';
 
 import * as which from 'which';
 
 import { ConfigurationConstants, DotnetConstants } from './constants';
 import Configuration from './configuration';
+import { Messages } from './ui/messages';
 
 const ListRuntimesArg = '--list-runtimes';
 const execFileAsync = promisify(execFile);
+const lstatAsync = promisify(fs.lstat);
 
-export async function hasSupportedDotnetVersion(): Promise<boolean> {
-  const dotnetExecutable = await getDotnetExecutablePath();
+// Returns an error message or undefined.
+export async function checkSupportedDotnetVersion(): Promise<string | undefined> {
+  const { path: dotnetExecutable, manual } = await getDotnetExecutablePath();
   try {
+    const stats = await lstatAsync(dotnetExecutable) as fs.Stats;
```

Comment:
nit: Instead of using `as`, you may want to parameterize `promisify`. Maybe something like this (untested):
```cs
const lstatAsync = promisify<string, fs.Stats>(fs.lstat);
```

Reply:
Oh actually the cast is not longer needed, I had added it because I was promisifying the wrong version fs.lstatAsync